docroot/modules/humsci/hs_courses/modules/hs_courses_importer/hs_courses_importer.install

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @file
+ * hs_courses_importer.install
+ */
+
+use Drupal\migrate\MigrateMessage;
+use Drupal\migrate_tools\MigrateExecutable;
+use Drupal\Core\Config\FileStorage;
+
+/**
+ * Rollback courses, revert the migration config.
+ */
+function hs_courses_importer_update_8001() {
+  $migrations = hs_field_helpers_migration_list();
+  /** @var \Drupal\migrate_plus\Entity\Migration $course_migration */
+  $course_migration = $migrations['hs_courses']['hs_courses'];
+  $log = new MigrateMessage();
+  $executable = new MigrateExecutable($course_migration, $log);
+  if (\Drupal::database()->schema()->tableExists('migrate_map_hs_courses')) {
+    $executable->rollback();
+  }
+}
+
+/**
+ * Drop the migrate table for courses.
+ */
+function hs_courses_importer_update_8002(){
```

Comment:
i added the course id & course code to `source.ids`. anytime the ids are added/deleted the table must be deleted so that it can be rebuilt with the additional columns on the next import. i could manually add the columns but this seemed easier. i'll add this comment in the update function

Reply:
Yes, please add the comment. Wasn't clear if you were navigating around a bug or doing it for convenience. Thanks.